src/clojure/puppetlabs/ssl_utils/core.clj

```diff
@@ -287,27 +291,42 @@
    :value    comment})
 
 (defn- create-authority-key-identifier
-  [public-key issuer-dn serial critical]
+  [public-key issuer-dn serial cert critical]
   {:oid      authority-key-identifier-oid
    :critical (boolean critical)
    :value    {:public-key    public-key
               :serial-number (if (number? serial) (biginteger serial))
-              :issuer-dn     issuer-dn}})
-
-(schema/defn ^:always-validate authority-key-identifier :- SSLExtension
-  "Create an `Authority Key Identifier` extension from a `PublicKey` object. The
-  extension created by this function is intended to be passed into
-  `sign-certificate` and `generate-certificate-request`, at which time the key's
-  hash will be computed and stored in the resulting object."
+              :issuer-dn     issuer-dn
+              :cert          cert}})
+
+(schema/defn ^:always-validate authority-key-identifier-options :- SSLExtension
+  "Create an `Authority Key Identifier` extension from a `PublicKey` object or
+   copy the existing `SubjectKeyIdentifier` from a certificate. This function is
+   intended to be passed into `sign-certificate` and `generate-certificate-request`.
+
+   PublicKey details:
+   The key's hash will be computed and stored when parsing the extensions list.
+
+   Certificate details:
+   Instead of using a `PublicKey` object, a certificate can be used. The
```

Comment:
PublicKey is capitalized so I assume this needs to be a Java object that is or implements `PublicKey`, is that correct? What is the equivalent java object or interface for "certificate".

Reply:
X509Certificate. Do you think that should be the schema for the argument?

Reply:
No, in fact I think it would be fine to remove the `:always-validate` metadata from this function. But I would use `X509Certificate` in the documentation.